braindecode/models/signal_jepa.py

```diff
@@ -0,0 +1,994 @@
+        for i, cl in enumerate(conv_layers_spec):
+            assert len(cl) == 3, "invalid conv definition: " + str(cl)
+            (dim, k, stride) = cl
+            conv_layers.append(
+                block(
+                    in_d,
+                    dim,
+                    k,
+                    stride,
+                    is_layer_norm=mode == "layer_norm",
+                    is_group_norm=mode == "default" and i == 0,
+                    conv_bias=conv_bias,
+                )
+            )
+            in_d = dim
+        self.emb_dim = dim  # last dim
+        self.conv_layers_spec = conv_layers_spec
+        self.cnn = nn.Sequential(*conv_layers)
+
+    def forward(self, batch):
+        """
+        Parameters
+        ----------
+            batch: dict with keys:
+
+                * X: (batch_size, n_chans, n_times)
+                    Batched EEG signal.
+
+        Returns
+        -------
+            local_features: (batch_size, n_chans * n_times_out, emb_dim)
+                Local features extracted from the EEG signal.
+                ``emb_dim`` corresponds to the ``dim`` of the last element of
+                ``conv_layers_spec``.
+        """
+        x = batch["X"]
+        batch_size, n_chans, n_times = x.shape
+        x = x.view(batch_size * n_chans, 1, n_times)
+        x: torch.Tensor = self.cnn(x)  # (batch_size * n_chans, emb_dim, n_times_out)
+        x = x.transpose(1, 2)
+        x = x.reshape(batch_size, -1, self.emb_dim)
+        return x  # (batch_size, n_chans * n_times_out, emb_dim)
+
+    @property
+    def receptive_fields(self):
+        rf = 1
+        receptive_fields = [rf]
+        for _, width, stride in reversed(self.conv_layers_spec):
+            rf = (rf - 1) * stride + width  # assumes no padding and no dilation
+            receptive_fields.append(rf)
+        return list(reversed(receptive_fields))
+
+    def n_times_out(self, n_times):
+        return n_times_out(self.conv_layers_spec, n_times)
+
+    def description(self, sfreq=None, n_times=None):
+        dims, _, strides = zip(*self.conv_layers_spec)
+        receptive_fields = self.receptive_fields
+        rf = receptive_fields[0]
+        desc = f"Receptive field: {rf} samples"
+        if sfreq is not None:
+            desc += f", {rf / sfreq:.2f} seconds"
+
+        ds_factor = math.prod(strides)
+        desc += f" | Downsampled by {ds_factor}"
+        if sfreq is not None:
+            desc += f", new sfreq: {sfreq / ds_factor:.2f} Hz"
+        desc += f" | Overlap of {rf - ds_factor} samples"
+        if n_times is not None:
+            n_times_out = self.n_times_out(n_times)
+            desc += f" | {n_times_out} encoded samples/trial"
+
+        n_features = [
+            f"{dim}*{rf}" for dim, rf in zip([1] + list(dims), receptive_fields)
+        ]
+        desc += f" | #features/sample at each layer (n_channels*n_times): [{', '.join(n_features)}] = {[eval(x) for x in n_features]}"
+        return desc
```

Comment:
```suggestion
```

we already have torchinfo, no?

Reply:
This gives more details (receptive field, downsampling factor, etc.)